plugins/statistics.koplugin/main.lua

```diff
@@ -1600,7 +1629,17 @@ function ReaderStatistics:getBookStat(id_book)
                 }
                 UIManager:show(self.kv)
             end,
-        }
+        },
+        { _("Average time per day"), datetime.secondsToClockDuration(user_duration_format, book_read_time/tonumber(total_days), false, true, true), separator = true },
+
+        -- Book progression
+        { _("Book start date"), datetime.secondsToDate(tonumber(first_open), true) },
+        { _("Last read page/Total pages"), string.format("%d / %d (%d%%)", last_page, pages, Math.round(100*last_page/pages)) },
+        { _("Last read date"), datetime.secondsToDate(tonumber(last_open), true), separator = true },
```

Comment:
May be flip these 2 "Last read" ones, so we have both "date" fields (Start & Last) near each other, instead of having that `1703 / 2338 (73%)` in between.

Reply:
I put those at the beginning and end to bookend (haha) this "progress" section, which mirrors the current statistics "progress" section too (which has "Current page/Total pages" and "Estimated time left" in the middle). It all goes back to this thinking: "Estimated time left" should come after "Book start date" but before "Estimated finish date". It would be strange to have:

- Book start date:                    Thu Jun 23 2022
- Estimate finish date:             Tue Dec 20 2022 (8 days)
- Estimated time left:               2h41m08s
- Current page/Total pages:   1543 // 1645 (94%)

If we really want to get rid of the 1703 / 2338 (73%) in between, we could do:

- Book start date:                    Thu Jun 23 2022
- Estimated time left:               2h41m08s
- Estimate finish date:             Tue Dec 20 2022 (8 days)
- Current page/Total pages:   1543 // 1645 (94%)

Reply:
This seems fine for the estimated stuff, where the start date is real, then some maths, then the unreal/estimated end and future date. For the current statistics, I don't need to grasp start and estimated as a pair. I would just jump to look at the start - or the estimated end.
For the past books, start and last date are both past, and they would usually be grasped as a pair. The nb of pages in between does not influence the last date, and feels a bit like noise preventing that pair grasping.

Reply:
Makes sense. Going to go with:

- **Book start date:** Thu Jun 23 2022
- **Estimated finish date:** Tue Dec 20 2022 (8 days)
- **Estimated time left:** 2h41m08s
- **Current page/Total pages:** 1543 // 1645 (94%)

and 

- **Book start date:** Thu Jun 23 2022
- **Last read date:** Mon Dec 12 2022
- **Last read page/Total pages:** 1703 / 2338 (73%)

Nicely dovetails and has the dates together.

Reply:
Maybe we should have the first say "Tue Dec 20 2022 (in 8 days)" and the second say "Mon Dec 12 2022 (today)" (or "Mon Dec 7 2022 (5 days ago)")

Reply:
> Maybe we should have the first say "Tue Dec 20 2022 (in 8 days)" and the second say "Mon Dec 12 2022 (today)" (or "Wed Dec 7 2022 (5 days ago)")

Could be indeed nice (dunno if we have yet "in/ago" datetime logic, and how nightmarish this would be to get right for translations :)

Reply:
Yes, see http://koreader.rocks/doc/modules/gettext.html#ngettext

Reply:
@Frenzie Isn't that only for a singular and a plural? What about a 0?

Reply:
Actually let's scrap the "today", "yesterday", "tomorrow", and just follow the nice simple "0 days ago", "1 day ago", "2 days ago" logic that `N_()` can handle

Reply:
> Isn't that only for a singular and a plural? What about a 0?

Whether 0 is treated differently depends on the language. If you mean to always treat it differently then that's a job for the programming language.